gtsam/basis/Chebyshev2.h

```diff
@@ -94,6 +102,16 @@ class GTSAM_EXPORT Chebyshev2 : public Basis<Chebyshev2> {
   static Weights CalculateWeights(size_t N, double x, double a = -1,
                                   double b = 1);
 
+  /**
+   * Calculate weights for all x in vector X.
+   * Returns M*N matrix where M is the size of the vector X,
+   * and N is the number of basis functions.
+   *
+   * Overriden for Chebyshev2.
+   */
+  static Matrix WeightMatrix(size_t N, const Vector& X, double a = -1,
```

Comment:
Where is this used? Just being worried about this MxN matrix being confused with the other MxN matrix that is used for VectorEvaluators. But I remember we talked about this as a speedup in some scenarios?

Reply:
This needs to be removed too.